### DISCUSSION TITLE
Are background and nodes layer stacked one above the other and some more question to bring more interactivity to the flow?

### DISCUSSION THREAD

**Discussion by ghost** [xyflow/xyflow#3014] [Q&A]
In this attached video, the nodes, background with id 1, background with id 2 are stacked one after the other. its evident that nodes are at the front, I have few questions
1. How the other background layers are stacked?
2. When I zoom in and out, how can I control so that I can see nodes not scaling, but only the background.
3. When I zoom in and out, how can I control so that I can see the background scaling, but nodes remaining same size?

https://user-images.githubusercontent.com/6542274/234104556-7a9f61da-3e55-4649-ad68-6b96fdbeb360.mov

**Comment by moklick** [comment]
1. Which other backgrounds? The backgrounds are stacked in the order as they appear in the DOM.
2. Why do you want to do that? What would be the reason for zooming?
3. same question as 2.?

**Reply by ghost** [reply]
Answer to #2 idea I have is to build a node based environment which creates a learning shape.

Please note my words would be vague, but let me try to put my visualisation as descriptive as it shall be.


I have an idea to create an interface for AI tech, just like how AI are math based, in my understanding, ai is a shape changing structure, which is a node in your project terms.


These nodes are exact form of a neuron, however most of the visualisation libraries I have seen tries to depict nodes in form of math, which can't be understood by many, instead I'm trying something alternative, in best words to describe is, all math equation drill down to shape, so I want to get rid of math, and illustrate in form of shape that changes its shape as it learns (memorizes input)



An experience I'd like to give to the use is that, they will create react-flow nodes with words, my background code will prompt or automatically the background code will convert the words into movements.

With respect to neural networks, there are two moving parts, the structure, the signal.


With these ideas, I wanted to help user to visualise the screen in two forms.


1. Nodes with words
2. Zoom out, it will take a form of a black and white neural network as in this form.
![Screenshot_20230430_175532_Chrome](https://user-images.githubusercontent.com/6542274/235352898-39a38b9e-b71f-4a52-9a1c-c68c8027dc5e.jpg)

**Reply by ghost** [reply]
My ask , I'll build nodes, I'll be able to zoom in and out, and at each level of zoom, the nodes take a different form


At low level user will see words in nodes and signals flowing through the edges, as the user zoomed out, it takes an electrifying view of an neuron or even the brain

**Reply by moklick** [reply]
not 100% sure if React Flow is the right tool here. However you can get the current zoom level inside a custom node via:

```js 
const zoom = useStore(s => s.transform[2])
```

and use that to change the content or form of the node. Example: https://reactflow.dev/docs/examples/interaction/contextual-zoom/
